src/use-swr.ts

```diff
@@ -55,7 +55,7 @@ const now = (() => {
 })()
 
 // setup DOM events listeners for `focus` and `reconnect` actions
-if (!IS_SERVER && window.addEventListener) {
+if (!IS_SERVER && window.addEventListener && document.addEventListener) {
```

Comment:
this will crash on deno and RN runtime cuz they don't have `document` as an accessible global reference.

maybe we can do like this? 🤔 

```js
if (!IS_SERVER) {
  const revalidate = () => ...
  
  const hasWindow = typeof window !== 'undefined'
  const hasDocument = typeof document !== 'undefined'
  
  if (hasDocument) {
     document.addEventListener(...)
  }
  if (hasWindow) {
     window.addEventListener(...)
     window.addEventListener(...)
  }
}
```

this might not be very elegant, feel free to share any other approaches

Reply:
I think your solution is intuitive and readable. I will apply it.

Reply:
By the way, I don't know RN development environment. 
So, I have one question. Does RN have document object? If not, I will add logic to deal with it.

Reply:
(+) I have another question on "this will crash on deno"
`deno` is not considered a server environment?
I mean `!IS_SERVER` doesn't mean `is not on deno environment`?

Reply:
oops, right deno is included in that server side condition.

Reply:
I see. I applied your solution on 5fd759c